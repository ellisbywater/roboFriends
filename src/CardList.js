import React from 'react'
import Card from './Card'

const CardList = ({robots}) => {
    return (
        <div>
            {robots.map(robot => {
                return <Card key={robot.email} name={robot.name} email={robot.email} imgUrl={robot.imgUrl} />
            })}
        </div>
    )
}

export default CardList