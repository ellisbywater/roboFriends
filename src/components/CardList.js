import React from 'react'
import Card from './Card'

const CardList = ({robots}) => {
    return (
        <div className="flex flex-wrap justify-center">
            {robots.map(robot => {
                return <Card key={robot.id} id={robot.id} name={robot.name} email={robot.email} />
            })}
        </div>
    )
}

export default CardList