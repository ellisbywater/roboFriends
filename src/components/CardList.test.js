import React from 'react'
import { shallow } from 'enzyme'
import CardList from './CardList'

it('expect to render to CardList component', () => {
    const mockRobots = [
        {
            id: 1,
            name: 'John Smith',
            email: 'smith@example.com'
        },
        {
            id: 2,
            name: 'Jana Ray',
            email: 'ray@example.com'
        }
    ]
    expect(shallow(<CardList robots={mockRobots} />)).toMatchSnapshot();
})