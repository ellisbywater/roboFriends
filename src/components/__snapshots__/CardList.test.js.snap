// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render to CardList component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    robots={
      Array [
        Object {
          "email": "smith@example.com",
          "id": 1,
          "name": "John Smith",
        },
        Object {
          "email": "ray@example.com",
          "id": 2,
          "name": "Jana Ray",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          email="smith@example.com"
          id={1}
          name="John Smith"
        />,
        <Card
          email="ray@example.com"
          id={2}
          name="Jana Ray"
        />,
      ],
      "className": "flex flex-wrap justify-center",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "email": "smith@example.com",
          "id": 1,
          "name": "John Smith",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "email": "ray@example.com",
          "id": 2,
          "name": "Jana Ray",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            email="smith@example.com"
            id={1}
            name="John Smith"
          />,
          <Card
            email="ray@example.com"
            id={2}
            name="Jana Ray"
          />,
        ],
        "className": "flex flex-wrap justify-center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "email": "smith@example.com",
            "id": 1,
            "name": "John Smith",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "email": "ray@example.com",
            "id": 2,
            "name": "Jana Ray",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
